Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/sysupdate/ui/OnlineUpdateProgressBarDialogFragment.java

```diff
@@ -0,0 +1,98 @@
+package org.dolphinemu.dolphinemu.features.sysupdate.ui;
+
+import android.app.Dialog;
+import android.app.ProgressDialog;
+import android.content.pm.ActivityInfo;
+import android.os.Bundle;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.fragment.app.DialogFragment;
+import androidx.lifecycle.ViewModelProvider;
+
+import org.dolphinemu.dolphinemu.R;
+
+public class OnlineUpdateProgressBarDialogFragment extends DialogFragment
+{
+  @NonNull
+  @Override
+  public Dialog onCreateDialog(@Nullable Bundle savedInstanceState)
+  {
+    // Rotating the device while the update is running can result in a title failing to import.
+    getActivity().setRequestedOrientation(ActivityInfo.SCREEN_ORIENTATION_LOCKED);
```

Comment:
This feels rather ugly in general. Is there really no better solution? (Not that I can think of what the better solution would be myself...)

Reply:
Unfortunately, the mitigation is the best thing I could come up with, I'm not sure why the update is stopping on rotation. :/